tests/testthat/test-geom-boxplot.R

```diff
@@ -38,6 +38,22 @@ test_that("can use US spelling of colour", {
   expect_equal(gpar$col, "#FF0000FF")
 })
 
+test_that("boxes with variable widths do not overlap", {
+  p <- ggplot(data = iris, aes(Species, Sepal.Length)) +
+    geom_boxplot(aes(colour = Sepal.Width < 3.2), varwidth = TRUE)
+  d <- layer_data(p)[c("xmin", "xmax")]
+  
+  overlaps <- vector(length = nrow(d) - 1)  
+  for (i in 2:nrow(d)) {
+    if (d$xmin[i] < d$xmax[i - 1]) {
```

Comment:
This for loop could become (with a small modification) a separate `group_overlapping` function, and you could use above.